fileicon: add page (#5953)
tldr-pages/tldr: pages/osx/fileicon.md

# fileicon

> > a macOS CLI for managing custom icons for files and folders, as a programmatic alternative to interactively using Finder.

Reviewer: ```suggestion
> A macOS CLI for managing custom icons for files and folders.
> A programmatic alternative to interactively using Finder.
```

Reviewer 2: This description seems a little too manual pagey...
Can it be simplified?

Reviewer 3: I think it could be succinctly put as
```suggestion
> macOS CLI to manage custom files/folder icons.
```
what do you guys think?

Reviewer: I'd prefer `file and folder icons`, but otherwise sounds good.

Reviewer: But linter doesn't accept if the description starts with a lowercase letter right?

Reviewer 3: Yeah, you're right.  In the long run that should be fixed, but I'll just put "A" back the beginning.

Suggestion updated.

> More information: <https://github.com/mklement0/fileicon>.

> - Assign custom icon derived from image file 'img.png' to file 'foo':
> 
> `fileicon set foo img.png`

Reviewer: ```suggestion
- Set a custom icon for a specific file or directory:

`fileicon set {{path/to/file_or_directory}} {{path/to/icon.png}}`
```
Does this command only accept PNG icons?

Reviewer 2: It probably takes all picture files, the documentation is no help since it uses .png as well... however it does suggest it can be changed to foo (without file extension) however this is probably redundant

Reviewer 2: Im going to leave it as png, the user will probably ignore it and it just gives an example picture, and if it does only accept .pngs, even better.

Its a win-win

Reviewer: @CleanMachine1 why did you resolve? I'd say my suggestion is still valid.

Reviewer 3: Yes, @marchersimon's suggestion would be better than the current one.

Reviewer 2: I thought I added it. I completely agree yours is better

> - Remove previously assigned custom icon from folder 'foodir':
> 
> `fileicon rm foodir`

Reviewer: ```suggestion
- Remove a custom icon from folder specific file or directory:

`fileicon rm {{path/to/file_or_directory}}`
```

Reviewer 2: > folder specific file or directory 

this doesn't make sense, grammatically.

Change it to something like

> from a specific file or directory

Reviewer 3: @marchersimon I think you meant to replace `folder` with `a`, so I fixed that ;)

> - Extract custom icon from file 'foo' to icon file 'foo.icns':
> 
> `fileicon get foo`

Reviewer: ```suggestion
- Save the custom icon of a file or directory as a `.icns` file into the current directory:

`fileicon get {{path/to/file_or_directory}}`
```

> - Test if folder 'foodir' has custom icon:
> 
> `fileicon test foodir`

Reviewer: ```suggestion
- Test if a specific file or directory has a custom icon:

`fileicon test {{path/to/file_or_directory}}`
```
